src/main/java/de/tum/cit/aet/artemis/core/service/CourseService.java

```diff
@@ -358,6 +359,7 @@ public Course findOneWithExercisesAndLecturesAndExamsAndCompetenciesAndTutorialG
         course.setNumberOfPrerequisites(prerequisiteRepository.countByCourse(course));
         // NOTE: in this call we only want to know if tutorial groups exist in the course, we will load them when the user navigates into them
         course.setNumberOfTutorialGroups(tutorialGroupRepository.countByCourse(course));
+        course.setFaqs(faqRepository.findAllByCourseIdAndFaqState(courseId, FaqState.ACCEPTED));
```

Comment:
This method is executed quite often. I suggest we only invoke `faqRepository.findAllByCourseIdAndFaqState` when faqs are activated in the course.